### ISSUE TITLE
Error: Call to a member function isCollection() on null

### ISSUE THREAD

**Issue by soyuka** [dunglas/DunglasApiBundle#257]
I'm not sure that this is directly related to DunglasApiBundle but thought it'd be nice to keep a trace here.

Error origin: [ItemNormalizer#denormalize](https://github.com/dunglas/DunglasApiBundle/blob/master/JsonLd/Serializer/ItemNormalizer.php#L244). Note that, compared to the `normalize` function, there is no test for `!$type`.
Anyway, type should not be empty here, so I tracked back this error to [AttributeMetadataFactory#getAttributeMetadataFor](https://github.com/dunglas/DunglasApiBundle/blob/master/Mapping/Factory/AttributeMetadataFactory.php#L57), where the `reflectionProperty` was empty.
 
Weird because this property is declared the exact same way as the other, and as a string:

```php
    /**
     * @var string
     *
     * @ORM\Column(name="name", type="string", length=255)
     */
    private $name;

    /**
     * @var string
     *
     * @ORM\Column(name="SIRET", type="string", length=255, nullable=true)
     */
    private $SIRET;
``` 

The bug came from `SIRET` declared in uppercase:

`AttributeName` is `sIRET` but the `reflectionClass` gives `SIRET`:

```php
ReflectionClass {#713
  constants: []
  properties: array:7 [
    "SIRET" => ReflectionProperty {#885
      +name: "SIRET"
      +class: "\\\InternalOrganization"
      modifiers: "private"
      extra: array:1 [
        "docComment" => """
          /**\n
               * @var string\n
               *\n
               * @ORM\Column(name="SIRET", type="string", length=255, nullable=true)\n
               */
          """
      ]
    }
  ]
}
```

I know properties should not be uppercased and I've changed it, but this could be a bug. I let y'all be the judge :).

**Comment by dunglas** [comment]
Need to be fixed.

**Comment by soyuka** [comment]
I could try, any idea where it should be fixed (maybe [here](https://github.com/dunglas/DunglasApiBundle/blob/master/Mapping/Loader/ReflectionLoader.php#L147))? In my opinion `reflectionClass` gives the correct attribute name.

**Comment by dunglas** [comment]
Yes it should do the trick.

**Comment by dunglas** [comment]
Closed by #271
